ramls/request-preference.json

```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Request preference schema",
+  "properties": {
+    "id": {
+      "description": "Unique request preference ID",
+      "type": "string"
+    },
+    "userId": {
+      "description": "UUID of user associated with this request preference",
+      "type": "string"
+    },
+    "holdShelf": {
+      "description": "Whether 'Hold shelf' option is available to the user.",
+      "type": "boolean"
+    },
+    "delivery": {
+      "description": "Whether 'Delivery' option is available to the user.",
+      "type": "boolean"
+    },
+    "defaultServicePointId": {
+      "description": "UUID of default service point for 'Hold shelf' option",
+      "type": "string"
+    },
+    "defaultDeliveryAddressTypeId": {
+      "description": "UUID of user's address type",
+      "type": "string"
+    },
+    "fulfillment": {
+      "description": "Preferred fulfillment type. Possible values are 'Delivery', 'Hold shelf'",
+      "type": "string",
+      "enum":["Delivery", "Hold shelf"]
+    },
+    "metadata": {
+      "description": "Metadata about creation and changes to request preference",
+      "$ref": "raml-util/schemas/metadata.schema",
+      "readonly": true
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "userId",
+    "holdShelf",
```

Comment:
holdShelf is required by default, is delivery required as well?

Reply:
yes, it has to defined also to understand whether delivery = true or false

Reply:
holdShelf is supposed to be always be true (i.e. UI checkbox is always checked), delivery can be false, but I think it still makes sense to always have this field in request